Extension/src/LanguageServer/client.ts

```diff
@@ -1294,20 +1299,34 @@ export class DefaultClient implements Client {
         this.rootFolder = workspaceFolder;
         this.rootRealPath = this.RootPath ? (fs.existsSync(this.RootPath) ? fs.realpathSync(this.RootPath) : this.RootPath) : "";
 
-        let storagePath: string | undefined;
+        const baseStoragePath: string | undefined = util.getBaseStoragePath();
+        let legacyStoragePath: string | undefined;
+        let storagePath: string | undefined = "";
+        let workspaceHash: string = "";
+
         if (util.extensionContext) {
-            const path: string | undefined = util.extensionContext.storageUri?.fsPath;
-            if (path) {
-                storagePath = path;
+            const fsPath: string | undefined = util.extensionContext.storageUri?.fsPath;
+            if (fsPath) {
+                workspaceHash = path.basename(path.dirname(fsPath));
+                legacyStoragePath = fsPath;
             }
         }
 
-        if (!storagePath) {
-            storagePath = this.RootPath ? path.join(this.RootPath, "/.vscode") : "";
+        if (!legacyStoragePath) {
+            legacyStoragePath = this.RootPath ? path.join(this.RootPath, ".vscode") : "";
         }
+
         if (workspaceFolder && vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1) {
-            storagePath = path.join(storagePath, util.getUniqueWorkspaceStorageName(workspaceFolder));
+            legacyStoragePath = path.join(legacyStoragePath, util.getUniqueWorkspaceStorageName(workspaceFolder));
+        }
+
+        this.legacyStoragePath = legacyStoragePath;
+        this.baseStoragePath = baseStoragePath;
+
+        if (baseStoragePath && workspaceHash) {
```

Comment:
Is `if (baseStoragePath && workspaceHash)` to check these are undefined or not empty?

Reply:
not empty

Reply:
Should it use `empty()` to check if the string is not empty?